token/driver/issue.go

```diff
@@ -12,10 +12,24 @@ import (
 	"github.com/hyperledger-labs/fabric-token-sdk/token/token"
 )
 
+// TokenConversionRequest is a request to convert tokens
+type TokenConversionRequest struct {
+	// Challenge is a challenge to be solved by the prover
+	Challenge []byte
+	// Tokens is a list of tokens to be converted
+	Tokens []token.LedgerToken
+	// Proof is a proof that the prover has solved the challenge
+	Proof []byte
+}
+
 // IssueOptions models the options that can be passed to the issue command
 type IssueOptions struct {
 	// Attributes is a container of generic options that might be driver specific
 	Attributes map[interface{}]interface{}
+	// TokenConversionRequest is a request to convert tokens
+	TokenConversionRequest *TokenConversionRequest
```

Comment:
Maybe we don't have to specify the reason that these tokens are there but the objective. That is, we could call them `TokenRedeemRequest`. The same for the interface `CheckRedeemProof` `GenRedeemProof` etc. This could make this field more versatile for other cases where we redeem some tokens to issue others transactionally (e.g. currency conversion).

Reply:
in this case, I think it is better to clarify the reason to avoid confusion. Different operations might have different requirements. Better to explicitly differentiate.